source/ur/ur.hpp

```diff
@@ -375,3 +375,12 @@ static inline void roundToHighestFactorOfGlobalSizeIn3d(
                MaxBlockDim[2]));
   roundToHighestFactorOfGlobalSize(ThreadsPerBlock[2], GlobalSize[2]);
 }
+
+template <typename T>
+inline bool listContainsElem(uint32_t numEventsInList, const T *phList,
```

Comment:
The previous `std::find` solution requires that use a vector, whereas this patch tries to avoid unnecessary allocations of vectors at each `urEnqueueKernelLaunch`